create_base_image/create_base_image_hostlib.sh

```diff
@@ -0,0 +1,222 @@
+  done
+}
+
+# Gets debian packages from Cuttlefish repo
+fetch_cf_package() {
+
+  local url="$1"
+  local branch="$2"
+  local repository_dir="${url/*\//}"
+  local debian_dir="$(basename "${repository_dir}" .git)"
+
+  git clone "${url}" -b "${branch}"
+  dpkg-source -b "${debian_dir}"
+  rm -rf "${debian_dir}"
+
+  # Gets cf-common version
+  for dsc in *dsc; do
+    CF_VER=$(basename "${dsc/*_/}" .dsc)
+    CF_VER="${CF_VER//\./-}"
+  done
+
+}
+
+# Gets build artifacts from Android Build API
+fetch_build_artifacts() {
+
+  local target="$1"
+  local build_id="$2"
+  
+  FETCH_ARTIFACTS=`mktemp`
+  curl "https://www.googleapis.com/android/internal/build/v3/builds/$build_id/$target/attempts/latest/artifacts/fetch_cvd?alt=media" -o $FETCH_ARTIFACTS
+
+  chmod +x $FETCH_ARTIFACTS
+  eval $FETCH_ARTIFACTS
+
+}
+
+
+main() {
+  set -o errexit
+  set -x
+
+  # SETUP AND SOURCE FILES EXTRACTION
+
+  # Flags setup
+  PZ=(--project=${FLAGS_build_project} --zone=${FLAGS_build_zone})
+  if [[ -n "${FLAGS_build_tags}" ]]; then
+    build_tags=("--tags=${FLAGS_build_tags}")
+  else
+    build_tags=()
+  fi
+
+  # Gets latest successful build_id from target branch in case no build_id is specified
+  if [[ -z "${FLAGS_build_id}" ]]; then
+    FLAGS_build_id=`curl "https://www.googleapis.com/android/internal/build/v3/builds?branch=$FLAGS_build_branch&buildAttemptStatus=complete&buildType=submitted&maxResults=1&successful=true&target=$FLAGS_build_target" 2>/dev/null | \
+    python2 -c "import sys, json; print json.load(sys.stdin)['builds'][0]['buildId']"`
+  fi
+
+  # Fetches cuttlefish common packages and target build artifacts
+  scratch_dir="$(mktemp -d)"
+  pushd "${scratch_dir}"
+    fetch_cf_package "${FLAGS_repository_url}" "${FLAGS_repository_branch}"
+    mkdir cuttlefish
+    pushd cuttlefish
+      fetch_build_artifacts "${FLAGS_build_target}" "${FLAGS_build_id}"
+    popd
+  popd
+  source_files=(
+    "create_base_image_gce.sh"
+    ${scratch_dir}/*
+  )
+
+  # Sets image and family names in case none were specified
+  FLAGS_build_target="${FLAGS_build_target//_/-}"
+  if [[ -z "${FLAGS_dest_image}" ]]; then
+    FLAGS_dest_image="halyard-${CF_VER}-${FLAGS_build_branch}-${FLAGS_build_target}-${FLAGS_build_id}"
+  fi
+  if [[ -z "${FLAGS_dest_family}" ]]; then
+    FLAGS_dest_family="halyard-${FLAGS_build_branch}-${FLAGS_build_target}"
+  fi
+
+  if [[ -n "${FLAGS_dest_family}" ]]; then
+    dest_family_flag=("--family=${FLAGS_dest_family}")
+  else
+    dest_family_flag=()
+  fi
+
+  # Deletes instances, disks and images with names that will be used for build
+  delete_instances=("${FLAGS_build_instance}" "${FLAGS_dest_image}")
+  gcloud compute instances delete -q \
+    "${PZ[@]}" "${delete_instances[@]}" || echo Not running
+  gcloud compute disks delete -q \
+    "${PZ[@]}" "${FLAGS_dest_image}" || echo No scratch disk
+  gcloud compute images delete -q \
+    --project="${FLAGS_build_project}" "${FLAGS_dest_image}"  || echo Not respinning
```

Comment:
It's ok to unconditionally delete the previous two, but for the image it may be a good idea to check that it doesn't exist instead and fail with some error message. If the caller really wants to replace the image then they should pass the `--respin` flag.

Reply:
So there would be a `--respin` boolean flag to see if the script should replace the current image?
If `"${FLAGS_respin}"` is false and the image exists, then it would fail and stop execution?

Reply:
exactly